docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -167,6 +167,105 @@ no dedicated array type, and any field could contain multiple values. The
 a specific order. See the mapping documentation on <<array, arrays>> for more
 background.
 
+[discrete]
+[[search-fields-nested]]
+==== Handling of nested fields
+
+The `fields` response for <<nested,`nested` fields>> is slightly different from that
+of regular object fields. While leaf values inside regular `object` fields are
+returned as a flat list, values inside `nested` fields are grouped according
```

Comment:
There may be some missing words here: "grouped according to maintain the independence" ?

Reply:
Probably I was thinking about an adverb like " values inside `nested` fields are grouped accordingly
to maintain the independence of each object inside the original nested array." But reading it again now I think just leaving it simply " values inside `nested` fields are grouped to maintain the independence of each object inside the original nested array.", wdyt as a native speaker? Does any of this make more sense?

Reply:
Ah I see now, 'accordingly' makes more sense. To me your second option is a bit clearer: "are grouped to maintain the independence..."